synapse/cortex.py

```diff
@@ -2943,10 +2951,11 @@ async def _onload():
                         await asyncio.sleep(0)
                 except asyncio.CancelledError:  # pragma: no cover
                     raise
-                except Exception:  # pragma: no cover
-                    logger.warning(f'onload failed for package: {name}')
+                except Exception as exc:  # pragma: no cover
+                    logger.warning(f'onload failed for package: {name}', exc_info=exc)
                 await self.fire('core:pkg:onload:complete', pkg=name)
-            self.schedCoro(_onload())
+
+            self.runActiveTask(_onload())
```

Comment:
I think this can only be called if you already confirm that we are active. might be worth adding logic to `runActiveTask()` to check and/or bail if self.activebase is None or something.

Reply:
I restored the check for `self.isactive`.